src/gui/activitywidget.cpp

```diff
@@ -400,6 +224,264 @@ void ActivityWidget::slotOpenFile(QModelIndex indx)
+        // will repeat the gui log.
+
+        // after one hour, clear the gui log notification store
+        if( _guiLogTimer.elapsed() > 60*60*1000 ) {
+            _guiLoggedNotifications.clear();
+        }
+        if( !_guiLoggedNotifications.contains(activity._id)) {
+            QString host = activity._accName;
+            // store the name of the account that sends the notification to be
+            // able to add it to the tray notification
+            // remove the user name from the account as that is not accurate here.
+            int indx = host.indexOf(QChar('@'));
+            if( indx>-1 ) {
+                host.remove(0, 1+indx);
+            }
+            if( !host.isEmpty() ) {
+                if( accNotified.contains(host)) {
+                    accNotified[host] = accNotified[host]+1;
+                } else {
+                    accNotified[host] = 1;
+                }
+            }
+            _guiLoggedNotifications.insert(activity._id);
+        }
+    }
+
+    // check if there are widgets that have no corresponding activity from
+    // the server any more. Collect them in a list
+    QList< Activity::Identifier > strayCats;
+    foreach( auto id, _widgetForNotifId.keys() ) {
+        NotificationWidget *widget = _widgetForNotifId[id];
+
+        bool found = false;
+        // do not mark widgets of other accounts to delete.
+        if( widget->activity()._accName != listAccountName ) {
+            continue;
+        }
+
+        foreach( auto activity, list ) {
+            if( activity.ident() == id ) {
+                // found an activity
+                found = true;
+                break;
+            }
+        }
+        if( ! found ) {
+            // the activity does not exist any more.
+            strayCats.append(id);
+        }
+    }
+
+    // .. and now delete all these stray cat widgets.
+    foreach( auto strayCatId, strayCats ) {
+        NotificationWidget *widgetToGo = _widgetForNotifId[strayCatId];
+        scheduleWidgetToRemove(widgetToGo, 0);
+    }
+
+    _ui->_notifyLabel->setHidden(  _widgetForNotifId.isEmpty() );
+    _ui->_notifyScroll->setHidden( _widgetForNotifId.isEmpty() );
+
+    int newGuiLogCount = accNotified.count();
+
+    if( newGuiLogCount > 0 ) {
+        // restart the gui log timer now that we show a notification
+        _guiLogTimer.restart();
+
+        // Assemble a tray notification
+        QString msg = tr("You received %n new notification(s) from %2.", "", accNotified[accNotified.keys().at(0)]).
+                arg(accNotified.keys().at(0));
+
+        if( newGuiLogCount >= 2 ) {
+            QString acc1 = accNotified.keys().at(0);
+            QString acc2 = accNotified.keys().at(1);
+            if( newGuiLogCount == 2 ) {
+                int notiCount = accNotified[ acc1 ] + accNotified[ acc2 ];
+                msg = tr("You received %1 new notifications from %2 and %3.").arg(notiCount).arg(acc1).arg(acc2);
```

Comment:
msg = tr("You received %n new notifications from %1 and %2.", notiCount).arg(acc1, acc2);

 - use the plural form even if it's always more than 1. There are languages in which the name are different for 2 or more than 10, or so.
 - use the arg with multiple argument as it otherwise might cause problem id there is a %1 in the name of one account.